cumulusci/robotframework/tests/salesforce/forms.robot

```diff
@@ -0,0 +1,102 @@
+*** Settings ***
+Resource        cumulusci/robotframework/Salesforce.robot
+Library         cumulusci.robotframework.PageObjects
+Library         Dialogs
+
+Suite Setup     Run keywords
+...  Initialize Test Objects
+...  Open test browser
+Suite Teardown  Delete records and close browser
+Force tags      forms
+
+*** Variables ***
+${account name}   ACME Labs
+${campaign name}  The Big SPAM
+
+*** Keywords ***
+Initialize test objects
+    [Documentation]  Create related objects used by this test
+
+    Require salesforce object  Account   name=${account name}
+    Require salesforce object  Campaign  name=${campaign name}
+
+Require Salesforce object
+    # We might want to make this available to everyone.
+    [Documentation]
+    ...  Create a salesforce object if it doesn't exist
+    ...
+    ...  Example:
+    ...
+    ...  | Require salesforce object  Account  name=ACME Labs  description=Created for test case
+
+    [Arguments]  ${object_type}  &{args}
+
+    ${result}=  salesforce query  ${object_type}  &{args}
+    # if we don't already have such an object, create one
+    Run keyword if  not $result
+    ...  Salesforce insert  ${object type}  &{args}
+    Run keyword if  not $result
+    ...  log  created object  DEBUG
+
+*** Test Cases ***
+Lightning based form - Opportunity
+    [Documentation]
+    ...  Sets all of the input fields for an opportunity, to make sure
+    ...  we at least support enough input field types to create an opportunity
+
+    [Setup]  Run keywords
+    ...  Go to page                  Home    Opportunity
+    ...  AND  Click Object Button    New
+    ...  AND  Wait for modal         New     Opportunity
+    [Teardown]   Click modal button  Cancel
+
+    Input form data
+    ...  Opportunity Name  The big one                             # required text field
+    ...  Account Name      ${Account Name}                         # lookup
+    ...  Amount            1b                                      # currency field
+    ...  Next Step         whatever                                # text field
+    ...  Close Date        4/01/2022                               # date field
+    ...  Private           checked                                 # checkbox
+    ...  Type              New Customer                            # combobox
+    ...  Stage             Prospecting                             # combobox
+    ...  Probability (%)   90                                      # percentage
+    ...  Description       this is a long description\nblah blah   # textarea
+    ...  Primary Campaign Source  ${Campaign Name}                 # combobox
+    ...  Lead Source       Purchased List                          # combobox
+
+    capture page screenshot
```

Comment:
Yes, but I think I'll leave it in. It's relatively harmless and might be useful if the keyword doesn't throw any errors but it's still not working right for some reason.

Reply:
I get a little nervous that at some point we're going to hit some file storage limit because of screenshots that we don't need, but it hasn't been a problem so far.